app/services/kyc_service.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+class KycService
+  # passport front + selfie OR driver license front and back + selfie IR id card front and back + selfie
+  REQUIRED_DOC_AMOUNT = { 'PASSPORT': 2, 'DRIVERS_LICENSE': 3, 'GOVERNMENT_ID': 3 }.freeze
+
+  class << self
+    def profile_step(profile)
+      user = profile.user
+      profile_label = user.labels.find_by(key: :profile)
+
+      if profile_label.nil? # first profile ever
+        user.labels.create(key: :profile, value: profile.state, scope: :private)
+      else
+        profile_label.update(value: profile.state) # re-submitted profile
+      end
+
+      return if Barong::App.config.kyc_provider == 'local' ||
+        profile.state == 'rejected' || profile.state == 'verified' || profile.state == 'drafted' # not an attempt for verification
+
+      KYC.const_get(Barong::App.config.kyc_provider.capitalize, false)::ApplicantWorker.perform_async(profile.id)
+    end
+
+    def document_step(document)
+      return if document.doc_type == 'Address'
+
+      user = document.user
+      user_document_label = user.labels.find_by(key: :document)
+
+      if user_document_label.nil? # first document ever
+        user.labels.create(key: :document, value: :pending, scope: :private)
+      else
+        user_document_label.update(value: :pending) # re-submitted document
```

Comment:
FIX NEEDED: we need to move this logic after check for `REQUIRED_DOC_AMOUNT`, as if enough docs will never come - we will get stucked in `pending`